supervisor/addons/configuration.py

```diff
@@ -0,0 +1,11 @@
+"""Confgiuration Objects for Addon Config."""
+
+from dataclasses import dataclass
+
+
+@dataclass(slots=True)
+class FolderMapping:
+    """Represent folder mapping configuration."""
+
+    path: str
```

Comment:
```suggestion
    path: str | None
```
The way these instances are created elsewhere in this PR the value for `path` may be `None`, the type should reflect that.